examples/gno.land/p/demo/memeland/memeland.gno

```diff
@@ -0,0 +1,196 @@
+		Author:        std.GetOrigCaller(),
+		Timestamp:     time.Unix(int64(t), 0),
+		Upvotes:       0,
+		UpvoteTracker: avl.NewTree(),
+	}
+
+	m.Posts.Set(id, newPost)
+
+	return id
+}
+
+func (m *Memeland) Upvote(id string) string {
+	post, exists := m.Posts.Get(id)
+	if !exists {
+		return "post with specified ID does not exist"
+	}
+
+	p, ok := post.(*Post)
+	if !ok {
+		return "error casting post"
+	}
+
+	caller := std.GetOrigCaller().String()
+
+	if _, exists := p.UpvoteTracker.Get(caller); exists {
+		return "user has already upvoted this post"
+	}
+
+	p.Upvotes += 1
+	m.Posts.Set(id, p)
+	p.UpvoteTracker.Set(caller, true)
+
+	return "upvote successful"
+}
+
+// GetPost - Retrieves a post
+func (m *Memeland) GetPost(id string) (*Post, bool) {
+	value, exists := m.Posts.Get(id)
+	if exists {
+		return value.(*Post), true
+	}
+	return nil, false
+}
+
+func (m *Memeland) GetPosts() []*Post {
+	var posts []*Post
+
+	m.Posts.Iterate("", "", func(key string, value interface{}) bool {
+		post := value.(*Post)
+		posts = append(posts, post)
+
+		return false
+	})
+
+	return posts
+}
+
+// GetPostsByTimestamp returns a JSON string of posts within the given timestamp range, supporting pagination.
+func (m *Memeland) GetPostsByTimestamp(startTimestamp, endTimestamp string, page, pageSize int) string {
+	var filteredPosts []*Post
+
+	if page < 0 {
+		return "page count cannot be less than 1"
+	}
+
+	if pageSize < 0 {
+		return "page size cannot be less than 1"
+	}
+
+	startIndex := (page - 1) * pageSize
+	endIndex := startIndex + pageSize
+	count := 0
+
+	// Convert timestamps to time.Time
+	startTime, err := strconv.Atoi(startTimestamp)
+
+	if err != nil {
+		return "could not parse start timestamp"
+	}
+	endTime, err := strconv.Atoi(endTimestamp)
+	if err != nil {
+		return "could not parse end timestamp"
+	}
+
+	start := time.Unix(int64(startTime), 0)
+	end := time.Unix(int64(endTime), 0)
+
+	// Use AVL tree traversal to filter posts by timestamp range
+	m.Posts.Iterate("", "", func(key string, value interface{}) bool {
+		post := value.(*Post) // Type assertion, assuming value is always *Post
+		if post.Timestamp.After(start) && post.Timestamp.Before(end) {
+			if count >= startIndex && count < endIndex {
+				filteredPosts = append(filteredPosts, post)
+			}
+			count++
+		}
+		return false
+	})
+
+	// Return JSON representation of filtered posts
+	return PostsToJSONString(filteredPosts)
+}
+
+// PostsToJSONString converts a slice of Post structs into a JSON string representation.
+func PostsToJSONString(posts []*Post) string {
+	var sb strings.Builder
+	sb.WriteString("[")
+
+	for i, post := range posts {
+		if i > 0 {
+			sb.WriteString(",")
+		}
+		sb.WriteString("{")
+		sb.WriteString(`"id":"` + post.ID + `",`)
+		sb.WriteString(`"data":"` + escapeString(post.Data) + `",`)
+		sb.WriteString(`"author":"` + escapeString(post.Author.String()) + `",`)
+		sb.WriteString(`"timestamp":"` + post.Timestamp.Format(time.RFC3339) + `",`)
```

Comment:
Why is the timestamp not formatted as unix?

Reply:
Now it is: [1121e74](https://github.com/gnolang/gno/pull/1751/commits/1121e7434d297c51d5cf418a46a4fb7b247b6be5)